manager/controlapi/task.go

```diff
@@ -104,10 +106,18 @@ func (s *Server) ListTasks(ctx context.Context, request *api.ListTasksRequest) (
 	if request.Filters != nil {
 		tasks = filterTasks(tasks,
 			func(e *api.Task) bool {
-				return filterContains(e.ServiceAnnotations.Name, request.Filters.Names)
+				name := e.Annotations.Name
+				if name == "" {
+					name = fmt.Sprintf("%v.%v", e.ServiceAnnotations.Name, e.Slot)
```

Comment:
I don't understand how this will work. If `Annotations.Name` was empty, how would the task have been returned by `FindTasks` above?